src/index.js

```diff
@@ -0,0 +1,178 @@
+        },
+    });
+
+    const { lat: latitude, lon: longitude } = response.data[0];
+    return { latitude, longitude };
+}
+
+const getWeather = async (coordinates) => {
+    const response = await axios.get('http://127.0.0.1:5000/weather', {
+        params: {
+            lat: coordinates["latitude"],
+            lon: coordinates["longitude"],
+        },
+    });
+
+    return response.data["main"]["temp"];
+};
+
+const loadControls = () => {
+    temperatureLabel = document.getElementById("tempValue");
+    temperatureDecreaseButton = document.getElementById("decreaseTempControl");
+    temperatureIncreaseButton = document.getElementById("increaseTempControl");
+    getRealtimeTemperatureButton = document.getElementById("currentTempButton");
+    landscapeImage = document.getElementById("landscape");
+    cityNameLabel = document.getElementById("headerCityName");
+    cityNameInput = document.getElementById("cityNameInput");
+    cityNameResetButton = document.getElementById("cityNameReset");
+    skyImage = document.getElementById("sky");
+    skySelectValue = document.getElementById("skySelect");
+    gardenBackground = document.getElementById("gardenContent");
+};
+
+const handleTemperatureDecreaseButtonClick = () => {
+    temperatureValue -= 1;
+    updateTemperature();
+};
+
+const handleTemperatureIncreaseButtonClick = () => {
+    temperatureValue += 1;
+    updateTemperature();
+};
+
+const handleCityNameInputChange = () => {
+    cityNameLabel.textContent = cityNameInput.value;
+};
+
+const handleCityNameResetButton = () => {
+    setDefaults();
+};
+
+const handleGetRealtimeTemperatureButtonClick = async () => {
+    const coordinates = await getCoordinates(cityNameInput.value);
+    const temperature = await getWeather(coordinates);
+    temperatureValue = Math.floor((temperature - 273.15) * 9/5 + 32);
+    updateTemperature();
+};
+
+const handleChangeSkySelect = () => {
+    skyImage.textContent = setSkyImage(skySelectValue.value);
+    gardenBackground.backgroundColor = setGardenBackgroundColor(skySelectValue.value);
+};
+
+const registerEvents = () => {
+    temperatureDecreaseButton.addEventListener("click", handleTemperatureDecreaseButtonClick);
+    temperatureIncreaseButton.addEventListener("click", handleTemperatureIncreaseButtonClick);
+    getRealtimeTemperatureButton.addEventListener("click", handleGetRealtimeTemperatureButtonClick);
+    cityNameResetButton.addEventListener("click", handleCityNameResetButton);
+    cityNameInput.addEventListener("input", handleCityNameInputChange);
+    skySelectValue.addEventListener("change", handleChangeSkySelect);
+};
+
+const setDefaults = () => {
+    temperatureValue = 72;
+    defaultCityName = 'Denver';
+    skySelectValue.value = 'Sunny';
+    showTemp();
+    showCity();
+    handleChangeSkySelect();
+    updateTemperature();
+}
+
+const onLoad = () => {
+    loadControls();
+    registerEvents();
+    setDefaults();
+    showTemp();
+    showCity();
+};
+
+onLoad();
```

Comment:
Consider registering the `onLoad` function to run when the `DOMContentLoaded` event occurs rather than calling it directly.